src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.Tests.Common.cs

```diff
@@ -154,6 +154,22 @@ public static async Task<Regex[]> GetRegexesAsync(RegexEngine engine, params (st
             RegexEngine.NonBacktrackingSourceGenerated => RegexOptionNonBacktracking | RegexOptions.Compiled,
             _ => throw new ArgumentException($"Unknown engine: {engine}"),
         };
+
+        /// <summary>Set the environment variable to the max value that effectively removes the check. Only used with Nonbacktracking engine.</summary>
+        public static void RemoveSafeSizeThresholdCheck()
```

Comment:
Rather than always increasing the limit to int.MaxValue, can we instead have this method take a `int maxSize` to set the max to?  Then we can use the smallest max possible for a given test and know if in the future something causes the test to start going above that previously-computed bound.

Reply:
Good point, will update, thought also occurred to me. So in the 6003 case could set it smth like 6005 smaller. Will need to recheck the numbers when I only count the leaves that more accurately reflect nr of NFA states also.